server/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -559,11 +560,14 @@ void phase1(IndexCommit snapshot, long startingSeqNo, IntSupplier translogOps, A
             if (canSkipPhase1(recoverySourceMetadata, request.metadataSnapshot()) == false) {
                 cancellableThreads.checkForCancel();
                 final boolean canUseSnapshots = useSnapshots && request.canDownloadSnapshotFiles();
+                Store.MetadataSnapshot targetMetadata = isSearchableSnapshotStore(shard.indexSettings().getSettings())
+                    ? recoverySourceMetadata
+                    : request.metadataSnapshot();
```

Comment:
this is essentially a quick way to ensure that no files are sent over when searchable snapshots are used, while still going through phase where cleanFiles is called at the end (and where we prepare the old data to be handled by the node)

Reply:
Can we add a comment? Otherwise it might be confusing at first to pass source == target to the recovery planner service.

I also think we should set `canUseSnapshots` to `false` as well in order to skip fetching the last snapshots infos? (I'm not familiar with this code, maybe it's done at the `StartRecoveryRequest` but I've seen nothing like this)

Reply:
Good point. I like the idea of setting canUseSnapshots to false, this will avoid unnecessarily reaching out to repo, and make this change more contained. I addressed this in 5a164e7.

Reply:
Thanks :+1: